snap/info.go

```diff
@@ -269,24 +269,25 @@ func (app *AppInfo) WrapperPath() string {
 }
 
 func (app *AppInfo) launcherCommand(command string) string {
-	securityTag := app.SecurityTag()
-	return fmt.Sprintf("/usr/bin/ubuntu-core-launcher %s %s %s", securityTag, securityTag, filepath.Join(app.Snap.MountDir(), command))
-
+	if app.Name == app.Snap.Name() {
+		return fmt.Sprintf("/usr/bin/snap run %s %s", command, filepath.Base(app.Name))
```

Comment:
We shouldn't need the filepath.Base here, I think? @kyrofa fixed it so that our app names are validated.